### PR TITLE
Adding Person, Student, and Teacher files

### PR DESCRIPTION
**Pull request by Abel-desu**
**In this project I create  **
- class `Person` with  the following:
  - Instance vars: `@id`, `@name`, and `@age`.
  - Constructor with `name`,  `age`, and `parent_permission` as parameter. `name` and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
  - Getters for `@id`, `@name`, and `@age`.
  - Setters for `@name` and `@age`.
  - Private method `is_of_age?` that returns `true` if `@age` is greater or equal to 18 and `false` otherwise.
  - Public method `can_use_services?` that returns `true` if person is of age or if they have permission from parents.

- class `Student` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
  - Method `play_hooky` that returns `"¯\(ツ)/¯"`.
-  class `Teacher` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
  - Override `can_use_services?` so it always returns `true`.
- Each class should be saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class Person
+  def initialize(_id, name, _age)
```
Comment (Ismailco) [inline person.rb]: - [ ] The Person class should have a `parent_permission` as optional and have a default value of `true`.
- [ ] The `name` parameter. should be optional and have a default value of `unknown`, kindly make sure to include these parameters so all the methods are useful

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  attr_writer :name
```
Comment (Ismailco) [inline person.rb]: - [ ]  `:name` and `:age` instance variables should be able to be modified, kindly consider using attr_accessor here for `:name`, and `:age`

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  def initialize(_id, name, _age)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = 10
+  end
+  attr_reader :id
+
+  def of_age?
+    @age >= 18
+  end
```
Comment (Ismailco) [inline person.rb:12]: - [ ] The `of_age?` method should be private, kindly consider moving the method down and adding `private` so the app is private

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @Abel-desu, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+class Person
+  def initialize(_id, _age, name = 'unknown')
```
Comment (HAMMAS-SALEEM) [inline person.rb]: - [ ] The Person class should have a **parent_permission** as optional and have a default value of **true** as requested by the previous reviewer. :blush:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @Abel-desu :wave:

## Highlights :1st_place_medal:

:heavy_check_mark: Following GitFlow
:heavy_check_mark: Descriptive PR Title
:heavy_check_mark: Descriptive PR Summary

Good job on implementing some of the changes requested by the previous reviewer, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_See Below Comments_

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi @Abel-desu , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**